Skuid_Techniques/Navigation/README.MD

```diff
@@ -0,0 +1,167 @@
+- Secondary dropdown navigation items.   
+- Remember my place. If the user navigates away from the single page app (opening a record detail page), and then hits the back button - open the section where they were last. Similarly, retain the correct context when user refreshes the page) 
+-  Data Driven navigation. Dynamically control navigation items shown based on some data we know.
+
+
+## Navigation: Two basic options 
+
+As is typical with Skuid - there are a number of ways you can build out this solution.  Each one has its pros and cons.  
+
+- The first option is to use the vertical navigation component - which is easier to set up - but not as ultimately powerful (it cannot support data driven use cases).  
+
+- The second option is to use one of Skuid’s data iteration components (list or deck).  This can provide all the complexity you need,  but will be more complicated to set up.
+
+Here we'll walk through those two major options and provide some example pages. 
+
+## Option 1: Navigation component. 
+
+There are two navigation components (vertical and horizontal).  Each are very similar in operation.  I’ll build out vertical nav.   These components cannot be bound to model data,  so you have to define all the navigation options in the page building process.  This does not allow you to build out completely dynamic navigation.  
+
+Here is how to handle the requirements listed above. 
+
+### Highlight the users current selection  
+First, configure the design system so the selected state actually is different.  Look for Selected State under the Navigation element node of your design system.  Change the background color, or text font… For your testing make it obvious. 
+
+The navigation components provide a standard selected state support.  But you can enhance this by using display logic (look for Selected conditions).  Maybe you want multiple entries selected, or you want to drive selection from another component - like a wizard button.   
+
+Note that selecting a second nav element will clear all Selected state conditions regardless of whether the selected state condition still is true.  (This is being reported as an enhancement)  
+
+### Display data in the navigation element 
+You can include model data in the labels of your navigation component.  Let users know how many open tasks there are before they open that navigation element. How many questions are still outstanding in that section of the survey. You can also use conditional merge syntax to show or hide HTML elements in the navigation labels based on underlying data.  Show that goofy smiley face icon if all tasks are complete.  
+
+Note that the navigation component is not connected to a model,  so you will have to use global merge syntax  `{{$Model.modelname.data.0.fieldname}}`.  In V2, components do not yet listen to changes of data through global merge syntax.  The indicators will not change unless you reload the page.  (Coming enhancement) 
```

Comment:
I can relate to wanting to call out "coming soon" features, but let's leave 'em out since our audience is here to get down to current-state business.

Reply:
Sure thing.